backends/fs/stdiostream.cpp

```diff
@@ -24,22 +24,74 @@
 // Disable symbol overrides so that we can use FILE, fopen etc.
 #define FORBIDDEN_SYMBOL_ALLOW_ALL
 
-// for Windows unicode fopen(): _wfopen()
-#if defined(WIN32) && defined(UNICODE)
+// for Windows unicode fopen(): _wfopen() and for MoveFileEx
+#if defined(WIN32)
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 #include "backends/platform/sdl/win32/win32_wrapper.h"
 #endif
 
 // Include this after windows.h so we don't get a warning for redefining ARRAYSIZE
 #include "backends/fs/stdiostream.h"
+#include "common/textconsole.h"
 
-StdioStream::StdioStream(void *handle) : _handle(handle) {
+StdioStream::StdioStream(void *handle) : _handle(handle), _path(nullptr) {
 	assert(handle);
 }
 
 StdioStream::~StdioStream() {
 	fclose((FILE *)_handle);
+
+	if (!_path) {
+		return;
+	}
+
+	// _path is set: recreate the temporary file name and rename the file to
+	// its real name
+	Common::String tmpPath(*_path);
+	tmpPath += ".tmp";
+
+	if (!moveFile(tmpPath, *_path)) {
+		warning("Couldn't save file %s", _path->c_str());
+	}
+
+	delete _path;
+}
+
+bool StdioStream::moveFile(const Common::String &src, const Common::String &dst) {
+#ifndef STDIOSTREAM_NO_ATOMIC_SUPPORT
+	// This piece of code can't in a subclass override as moveFile is called from the destructor.
+	// In this case, the vtable is reset to StdioStream one before calling moveFile.
```

Comment:
```suggestion
	// In this case, the vtable is reset to StdioStream once before calling moveFile.
```

Reply:
`one` was the word really intended. :)
I added a `the` before `StdioStream` to make it clearer.